### PR TITLE
Fixes VSTS Bug 672522: Need Performance improvement on "Search in

### PR DESCRIPTION
**Pull request by mkrueger**
Solution" / "Scrolling over search results" observing ~9 seconds delay

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/672522

Fixes VSTS Bug 682954: Search result text height differences

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs [hunk 1]
```diff
@@ -165,43 +162,65 @@ void CreateMarkup (SearchResultWidget widget, TextEditor doc, Editor.IDocumentLi
...
-						color1.L -= 0.5;
-					} else {
-						color1.L += 0.5;
+			var tabSize = doc.Options.TabSize;
+			this.markup = this.selectedMarkup = markup = FormatMarkup (Ambience.EscapeText (lineText), trimStart, trimEnd, tabSize);
+
+			var searchColor = GetBackgroundMarkerColor (widget.HighlightStyle);
+
+			var selectedSearchColor = widget.Style.Base (Gtk.StateType.Selected);
+			selectedSearchColor = searchColor.AddLight (-0.2);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs]: This color could be cached and brightness could be cached.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs]: Afaik, HslColor stuff is slow, that's why I'm suggesting

Comment (mkrueger) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs]: It's all done lazy so we're speaking of maybe ~20 calls here on scrolling.
Optimize only when it's a problem.

**Comment by slluis** [comment]
Please add a unit test for the fix of 682954.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs (hunk 1 above)
Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResult.cs]: :+1: was just making sure

**Comment by mkrueger** [comment]
Added unit test.

**Comment by slluis** [comment]
@monojenkins backport release-7.6

**Comment by mkrueger** [comment]
@monojenkins backport release-7.7

**Pull request merged**